lib/extract_field/FieldExtractor.js

```diff
@@ -5,32 +5,32 @@ const SentThrough = { BODY: 'body', QUERY_PARAM: 'query-param', HEADER: 'header'
 const ContentType = { JSON: 'application/json', URL_ENCODED: 'application/x-www-form-urlencoded', MULTIPART_FORM: 'multipart/form-data' };
 
 class FieldExtractor {
-    constructor(sentThrough, fieldsToExtract) {
+    constructor(sentThrough, extractionCallback, fieldsToExtract) {
```

Comment:
what is extractionCallback? the success indicator? I think we need a more meaningful name

Reply:
A custom call back to extract the credentials at a particular endpoint if the configurations of `user_field`, `pass_field`, and `sent_through` aren't enough. For example, if they pass JSON:
```JSON
{ "credentials": "username=test@test.com&password=1234" }
```
Which was actually the case for TextNow